x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/MatchersUtils.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.regex.Pattern;
+
+/**
+ * Utility class containing custom hamcrest {@link Matcher} implementations or other utility functionality related to hamcrest.
+ */
+public class MatchersUtils {
+
+    /**
+     * Custom matcher implementing a matcher operating on json strings ignoring whitespaces, which are not inside a key or a value.
+     *
+     * Example:
+     * {
+     *     "key": "value"
+     * }
+     *
+     * will match
+     *
+     * {"key":"value"}
+     *
+     * as both json strings are equal ignoring the whitespace, which does not reside in a key or a value.
+     *
+     */
+    protected static class IsEqualIgnoreWhitespaceInJsonString extends TypeSafeMatcher<String> {
+
+        protected static final Pattern WHITESPACE_IN_JSON_EXCEPT_KEYS_AND_VALUES_PATTERN = createPattern();
+
+        private static Pattern createPattern() {
+            String regex = "(?<=[:,\\[{])\\s+|\\s+(?=[\\]}:,])|^\\s+|\\s+$";
```

Comment:
nit: I would prefer if the meaning of this regex pattern were a little closer to the definition of the pattern. Perhaps this `regex` variable can be renamed to `WHITESPACE_IN_JSON_EXCEPT_KEYS_AND_VALUES_PATTERN` or something? 

Right now, nothing within this `createPattern` function indicates what it does.

Reply:
Fair point! I'll address the nits in a follow-up PR so I can directly continue with embeddings